code/lib/cli/src/automigrate/fixes/nextjs-framework.ts

```diff
@@ -0,0 +1,194 @@
+  const nextjsAddon = addons?.find((addon) =>
+    typeof addon === 'string'
+      ? addon === 'storybook-addon-next'
+      : addon.name === 'storybook-addon-next'
+  );
+
+  if (!nextjsAddon || typeof nextjsAddon === 'string') {
+    return {};
+  }
+
+  return nextjsAddon.options || {};
+};
+
+/**
+ * Does the user have a nextjs project but is not using the @storybook/nextjs framework package?
+ *
+ * If so:
+ * - Remove the dependencies if webpack (@storybook/react-webpack5)
+ * - Install the nextjs package (@storybook/nextjs)
+ * - Uninstall existing legacy addons: storybook-addon-next and storybook-addon-next-router
+ * - Update StorybookConfig type import (if it exists) from react-webpack5 to nextjs
+ * - Update the main config to use the new framework
+ * -- removing legacy addons: storybook-addon-next and storybook-addon-next-router
+ * -- moving storybook-addon-next options into frameworkOptions
+ */
+export const nextjsFramework: Fix<NextjsFrameworkRunOptions> = {
+  id: 'nextjsFramework',
+
+  async check({ packageManager }) {
+    const packageJson = packageManager.retrievePackageJson();
+    const allDeps = { ...packageJson.dependencies, ...packageJson.devDependencies };
+
+    if (!allDeps.next) {
+      return null;
+    }
+
+    const { mainConfig, version: storybookVersion } = getStorybookInfo(packageJson);
+    if (!mainConfig) {
+      logger.warn('Unable to find storybook main.js config, skipping');
+      return null;
+    }
+
+    const storybookCoerced = storybookVersion && semver.coerce(storybookVersion)?.version;
+    if (!storybookCoerced) {
+      logger.warn(dedent`
+        ❌ Unable to determine storybook version, skipping ${chalk.cyan('nextjsFramework')} fix.
+        🤔 Are you running automigrate from your project directory?
+      `);
+      return null;
+    }
+
+    if (!semver.gte(storybookCoerced, '7.0.0')) {
+      return null;
+    }
+
+    const main = await readConfig(mainConfig);
+
+    const frameworkPackage = main.getFieldValue(['framework']);
+
+    if (!frameworkPackage) {
+      return null;
+    }
+
+    const frameworkPackageName =
+      typeof frameworkPackage === 'string' ? frameworkPackage : frameworkPackage.name;
+
+    if (frameworkPackageName === '@storybook/react-vite') {
+      logger.info(dedent`
+        We've detected you are using Storybook in a Next.js project.
+
+        In Storybook 7, we introduced a new framework package for Next.js projects: @storybook/nextjs.
+
+        This package provides a better experience for Next.js users, however it is only compatible with the webpack 5 builder, so we can't automigrate for you, as you are using the Vite builder.
+        
+        If you are interested in using this package, see: ${chalk.yellow(
+          'https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#nextjs-framework'
+        )}
+      `);
+
+      return null;
+    }
+
+    // we only migrate from react-webpack5 projects
```

Comment:
Given that `@storybook/react-webpack5` is only available in `7.0+`, It seems _far_ more likely that someone would be upgrading from a project with `@storybook/react` & `@storybook/builder-webpack5`. Perhaps, for this scenario, we could leverage the fix [tested here](https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/automigrate/fixes/new-frameworks.test.ts#L132-L172), and _then_ re-run this fix?